File: 1256-rank-transform-of-an-array/1256-rank-transform-of-an-array.java
import java.util.Arrays;
import java.util.HashMap;

public class Solution {
    public int[] arrayRankTransform(int[] arr) {
        int[] temp = Arrays.copyOf(arr, arr.length);  
        Arrays.sort(temp);  
        
        HashMap<Integer, Integer> rankMap = new HashMap<>();  
        int rank = 0;
        
        for (int i = 0; i < temp.length; i++) {
            if (!rankMap.containsKey(temp[i])) {
                rank++;
                rankMap.put(temp[i], rank);  
            }
        }

        int[] result = new int[arr.length];
        for (int i = 0; i < arr.length; i++) {
            result[i] = rankMap.get(arr[i]);  
        }

        return result;
    }

}

File: 1256-rank-transform-of-an-array/1256-rank-transform-of-an-array.py
import copy
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        temp = copy.deepcopy(arr) 
        temp.sort()
        d={}
        prev = -10^9 - 1
        x = 0 
        for i in (temp):
            if i not in d :
                x +=1
                d[i] = x 
        ans = []
        for i in range(len(arr)):
            ans.append(d[arr[i]]) 
        return ans  
